interfaces/Portalicious/src/app/components/query-table/components/query-table-column-management/query-table-column-management.component.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  input,
+  model,
+  output,
+} from '@angular/core';
+import {
+  FormControl,
+  FormGroup,
+  FormsModule,
+  ReactiveFormsModule,
+} from '@angular/forms';
+
+import { injectMutation } from '@tanstack/angular-query-experimental';
+import { ButtonModule } from 'primeng/button';
+import { CheckboxModule } from 'primeng/checkbox';
+
+import { FormSidebarComponent } from '~/components/form/form-sidebar.component';
+import { QueryTableColumn } from '~/components/query-table/query-table.component';
+
+@Component({
+  selector: 'app-query-table-column-management',
+  standalone: true,
+  imports: [
+    ButtonModule,
+    FormSidebarComponent,
+    CheckboxModule,
+    ReactiveFormsModule,
+    FormsModule,
+  ],
+  templateUrl: './query-table-column-management.component.html',
+  styles: ``,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class QueryTableColumnManagementComponent<
+  TData extends { id: PropertyKey },
+> {
+  columns = input.required<QueryTableColumn<TData>[]>();
+  visibleColumns = model.required<QueryTableColumn<TData>[]>();
+  readonly resetColumnVisibility = output();
+
+  formVisible = model<boolean>(false);
+
+  formGroup = new FormGroup({
+    selectedColumns: new FormControl<QueryTableColumn<TData>[]>([], {
+      nonNullable: true,
+    }),
+  });
+
+  showColumnManagement() {
+    this.formGroup.patchValue({
+      selectedColumns: this.visibleColumns(),
+    });
+    this.formVisible.set(true);
+  }
+
+  updateColumnVisibility = injectMutation(() => ({
+    // We don't technically need a mutation here, but we're using one
+    // so that we can easily reuse the form-sidebar component
+    mutationFn: () => Promise.resolve(),
+    onSuccess: () => {
+      this.visibleColumns.set(this.formGroup.getRawValue().selectedColumns);
+      this.formVisible.set(false);
+    },
+  }));
```

Comment:
Yeah this is kind of a fake mutation...

Maybe a better way to write this would be this now that I take another look at it:

```ts
updateColumnVisibility = injectMutation(() => ({
    // We don't technically need a mutation here, but we're using one
    // so that we can easily reuse the form-sidebar component
    mutationFn: () => Promise.resolve(this.formGroup.getRawValue().selectedColumns),
    onSuccess: (selectedColumns) => {
      this.visibleColumns.set(selectedColumns);
      this.formVisible.set(false);
    },
  }));
  ```
  
  But it doesn't really matter. I'm just using a mutation "for pretend" because we aren't actually mutating anything on the backend.

Reply:
I update this function with the snippet I shared above. Let me know if you see other improvements.